src/abstracts/SablierV2MerkleStreamer.sol

```diff
@@ -35,6 +35,9 @@ abstract contract SablierV2MerkleStreamer is
     /// @inheritdoc ISablierV2MerkleStreamer
     bytes32 public immutable override merkleRoot;
 
+    /// @inheritdoc ISablierV2MerkleStreamer
+    bool public immutable override transferable;
```

Comment:
We need to define this before `merkleRoot` to save gas by tight packing.

Reply:
Update: it is not true that storage variables are tightly packed like struct fields.

- https://chat.openai.com/share/67619484-f686-4d11-8cac-cb9e8d4ca1ce
- https://ethereum.stackexchange.com/q/155029/24693

Reply:
ChatGPT seems to have given an incorrect response here. Storage variables are also tightly packed.

It's true that because of dealing with reduced-size variables in the same slot, changing the value of `y` will consume more gas (~100 in comparison) in the case of contract B, but it will also save gas (~16,000 in comparison) when setting it the first time. It depends on the use case and how many times each variable is read or written.

However, in the above code, it does not matter because immutable variables are part of the creation code and not of the contract storage.

- https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html
- https://docs.soliditylang.org/en/latest/contracts.html#immutable